runelite-client/src/main/java/net/runelite/client/plugins/woodcutting/WoodcuttingPlugin.java

```diff
@@ -190,11 +205,31 @@ public void onChatMessage(ChatMessage event)
 
 			if (event.getMessage().contains("A bird's nest falls out of the tree") && config.showNestNotification())
 			{
-				notifier.notify("A bird nest has spawned!");
+				if (clueTierSpawned == null || clueTierSpawned.getTier() >= config.clueNestNotifyTier().getTier())
+				{
+					notifier.notify("A bird nest has spawned!");
+				}
+				// Clear the clue tier that has previously spawned
+				clueTierSpawned = null;
 			}
 		}
 	}
 
+	@Subscribe
+	public void onItemSpawned(ItemSpawned itemSpawned)
+	{
+		if (CLUE_NEST_IDS.contains(itemSpawned.getItem().getId()))
+		{
+			// It seems that falling nests spawn before the chat message is displayed, thus we can save state here and notify on game message.
+			clueTierSpawned = clueNestIdToTier.get(itemSpawned.getItem().getId());
+		}
+		else if (NON_CLUE_NEST_IDS.contains(itemSpawned.getItem().getId()))
+		{
+			// Clear the tier spawned to prevent player nest drop from affecting notification.
```

Comment:
How would a player dropped nest affect the notification? A nest has to spawn on the same tick the "A bird's nest falls out of the tree" chat message is received in order to affect the tier.

Reply:
So, imagine we don't set `clueTierSpawned` to `null` here, and consider the following scenario:
1. Player sets clue tier notification to `Hard`
2. Player gets a beginner clue nest, doesn't get notified (which is fine), picks it up and then decides to drop it.
3. Upon this drop, `onItemSpawned` is invoked with the beginner clue nest itemId, causing `clueTierSpawned` to be equal `ClueNestTier.BEGINNER`
4. Now, imagine some regular seed bird nest is dropped. `onItemSpawned` will be invoked but without setting `clueTierSpawned` to `null`, so it'll still be `ClueNestTier.BEGINNER`.
5. "A bird's nest falls out of the tree" chat message will be then received, and the check ```if (clueTierSpawned == null || clueTierSpawned.ordinal() >= config.clueNestNotifyTier().ordinal())``` will not pass, since `clueTierSpawned` is not `null` and the ordinal of it is for `BEGINNER` which is less than the ordinal for `HARD` as set in config.

Reply:
Your proposed scenario is not accurate as step 4 would reach this code branch and set the spawned tier to `null`.

Reply:
Forgive me, I thought that you wanted me to clarify why my comment exists; it does because it explains why the line below it is necessary. Which, it is - if you imagine this code branch (setting the spawned tier to `null`) does not exist, then a player dropping a clue nest could trigger my scenario and introduce a bug